### ISSUE TITLE
Cannot build buildable lib that depends on other buildable lib

### ISSUE THREAD

**Issue by leon** [nrwl/nx#6229]
## Current Behavior
I'm getting an error when building the lib that depends on the first lib.

```
File '/Users/leon/Temp/labs-buildable-libs/libs/shared/domain/src/index.ts' is not under 'rootDir' 'libs/shared/auth'. 'rootDir' is expected to contain all source files.
import { User } from '@myorg/shared/domain';
```

it seems that even though nx knows the correct dependency, (nx dep-graph)
the builder for the lib does not take into account that the other lib is buildable.

## Expected Behavior
I should be able to build each library separately so I can use nx cloud to cache my libs.

## Steps to Reproduce
Here is a repo with instructions and a very simple workspace to illustrate the problem.
https://github.com/leon/labs-nx-buildable-libs

The repo is very light to just illustrate the problem.
in my real repo we have multiple angular apps / nest.js apis and loads of workspace:library libs.
The whole point of buildable libs for us will be to pre compute a lot of these libs to get a faster build.
we use a lot of shared domain models that are skattered across the libs, and they sometimes extend each other, so we need to be able to build a lib, that then references the other lib.

### Environment
```
 Node : 14.17.2
  OS   : darwin x64
  yarn : 1.22.10
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.5.1
  @nrwl/cypress : Not Found
  @nrwl/devkit : 12.5.1
  @nrwl/eslint-plugin-nx : 12.5.1
  @nrwl/express : Not Found
  @nrwl/jest : 12.5.1
  @nrwl/linter : 12.5.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.5.1
  @nrwl/web : Not Found
  @nrwl/workspace : 12.5.1
  @nrwl/storybook : Not Found
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

**Comment by leon** [comment]
I also tried generating two `@nrwl/angular:library` libs where one depends on the other. but it's the same problem there.

https://github.com/leon/labs-nx-buildable-libs/tree/angular

```
nx build ng-nglib1
```
works

```
nx build ng-nglib2
```
fails with 

```
libs/ng/nglib1/src/lib/lib1.model.ts:1:1 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/lib/lib1.model.ngtypecheck.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

1 export interface Lib1Model {
  
libs/ng/nglib1/src/lib/ng-nglib1.module.ts:1:1 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/lib/ng-nglib1.module.ngtypecheck.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

1 import { NgModule } from '@angular/core';
  
libs/ng/nglib1/src/index.ts:1:1 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/index.ngtypecheck.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

1 export * from './lib/lib1.model';
  
libs/ng/nglib1/src/index.ts:1:15 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/lib/lib1.model.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

1 export * from './lib/lib1.model';
                ~~~~~~~~~~~~~~~~~~
libs/ng/nglib1/src/index.ts:2:15 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/lib/ng-nglib1.module.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

2 export * from './lib/ng-nglib1.module';
                ~~~~~~~~~~~~~~~~~~~~~~~~
libs/ng/nglib2/src/lib/lib2.model.ts:1:27 - error TS6059: File '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib1/src/index.ts' is not under 'rootDir' '/Users/leon/Temp/labs-buildable-libs/libs/ng/nglib2/src'. 'rootDir' is expected to contain all source files.

1 import { Lib1Model } from '@myorg/ng/nglib1';
```

**Comment by TomTomB** [comment]
Interesting sidenote: The issue only appears, if the libs are inside subfolders. 

Example
`lib-two` depends on `lib-one`

This folder structure will build
``` 
libs/lib-one
libs/lib-two
``` 

And this will produce the error 
``` 
libs/foo/lib-one
libs/lib-two
```

**Comment by xapou** [comment]
I encounter the exact same issue with libs inside sub folders.

```json
"@nrwl/cli": "12.5.2",
```

I did not try to flatten everything though.

**Comment by simoneatum** [comment]
Did someone found a workaround? This is a blocking issue for us and i haven't found a workaround to build the project... 
Thanks your help!

**Comment by simoneatum** [comment]
> Did someone found at least a workaround? This is a blocking issue for us and i wasn't able to find a solution to build the project...
> Thanks for your help!

update: I was able to resolve my issue updating nx to 12.5.9, removing the libraries that have problem and recreating them with the angular library schematic instead of the workspace schematic

**Comment by leosvelperez** [comment]
Thanks for reporting this!

There are 2 issues being mentioned in this thread:

1. Workspace buildable libraries don't build when having other workspace libraries as dependencies. This was resolved in #6475 and it should be published in the next patch version (12.6.1). Please keep an eye on the releases to see when the fix is available.

2. Angular buildable libraries don't build when created inside a directory and with other Angular libraries as dependencies. This was resolved in #6429 and was already published in [Nx 12.6.0](https://github.com/nrwl/nx/releases/tag/12.6.0).

**Comment by leosvelperez** [comment]
This is now released, you can use 12.6.2 to get the first issue mentioned in the comment above resolved.

I'm closing this one. Please feel free to open another issue if you still have problems with this.

**Comment by TomTomB** [comment]
It's probably worth metioning that changing a lib's name (inside package.json) to a valid npm name results in the same error. 

Example:
`"name": "@my-org/store/auth",`  - this will build (but shows a warning inside VSCode)
`"name": "@my-org/store-auth",`  - this will throw the error(s)
